### PR TITLE
Add and verify `weak-subjectivity-checkpoint` flag

### PR DESCRIPTION
**Pull request by terencechain**
**What type of PR is this?**

> Feature

**What does this PR do? Why is it needed?**
This PR implemented a CLI flag to take in weak subjectivity checkpoint. As discussed with teams and in the reference (below). The CLI input is in `block_root:epoch_number` format, where `block_root` is the root of a block at `epoch_number` is the epoch of the block. 

It also added a helper to validate and parse the input string + tests to go with it. The results are inserted to initial sync service struct but not used at the moment. It will be used in the subsequent PRs

Reference: https://github.com/ethereum/eth2.0-specs/blob/weak-subjectivity-guide/specs/phase0/weak-subjectivity.md#weak-subjectivity-sync-procedure

### CODE REVIEW

File: beacon-chain/node/helper.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+	// Get the hexadecimal block root from input string.
+	s := strings.Split(wsp, ":")
+	bRoot, err := hex.DecodeString(s[0])
+	if err != nil {
+		return nil, 0, err
+	}
+
+	// Get the epoch number from input string.
+	epoch, err := strconv.ParseUint(s[1], 10, 64)
```
Comment (rauljordan) [inline beacon-chain/node/helper.go:35]: what if len is < 2?

Comment (terencechain) [reply beacon-chain/node/helper.go:35]: Good point. I'll add a check

File: beacon-chain/node/helper_test.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+	}{
+		{
+			name:    "No column in string",
+			input:   "0x111111;123",
+			wantErr: true,
+			errStr:  "did not contain column",
+		},
+		{
+			name:    "Correct input #1",
+			input:   "0x010203:987",
```
Comment (nisdas) [inline beacon-chain/node/helper_test.go:27]: shouldnt the block root be 32 bytes. Lets add a check for it.

File: beacon-chain/node/helper.go [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+	// Strip prefix "0x" if it's part of the input string.
+	if strings.HasPrefix(wsp, "0x") {
+		wsp = wsp[2:]
+	}
+
+	// Get the hexadecimal block root from input string.
+	s := strings.Split(wsp, ":")
+	if len(s) != 2 {
+		return nil, 0, errors.New("bad format string")
+	}
```
Comment (rauljordan) [inline beacon-chain/node/helper.go:32]: Maybe output why it is not well formatted, showing them the expected format. Perhaps that should also be part of the flag usage if not specified already

Comment (terencechain) [reply beacon-chain/node/helper.go:32]: it's part of the flag usage. I'll reiterate in the error again

**Review by rauljordan — APPROVED** [review]

**Pull request merged**